lib/parallel.rb

```diff
@@ -337,10 +337,11 @@ def physical_processor_count
       end
     end
 
-    # Number of processors seen by the OS, used for process scheduling
+    # Number of processors seen by the OS or value considering CPU quota if the process is inside a cgroup,
+    # used for process scheduling
     def processor_count
-      require 'etc'
-      @processor_count ||= Integer(ENV['PARALLEL_PROCESSOR_COUNT'] || Etc.nprocessors)
+      require 'concurrent-ruby'
```

Comment:
how about this: no new dependency or version conflicts with other concurrent-ruby usage
... and maybe a readme entry for "if you want x add concurrent-ruby"
```
begin
  require 'concurrent-ruby'
  Concurrent.available_processor_count.to_i
rescue LoadError
  require 'etc'
  Etc.nprocessors
end
```

Reply:
maybe also use `.floor` instead of `.to_i` to make this more explicit

Reply:
Thanks for your review.

> how about this: no new dependency or version conflicts with other concurrent-ruby 

This makes sense to me. I've updated my fix to use this approach.